core/src/utils/prove_distributed/checkpoints.rs

```diff
@@ -0,0 +1,154 @@
+use std::sync::{mpsc::SyncSender, Arc};
+
+pub use crate::{air::PublicValues, runtime::Program, stark::RiscvAir};
+
+use crate::{
+    runtime::{ExecutionRecord, NoOpSubproofVerifier, Runtime},
+    stark::{MachineProver, MachineRecord},
+    utils::{baby_bear_poseidon2::Val, BabyBearPoseidon2, SP1CoreOpts},
+};
+
+use super::Checkpoint;
+
+fn trace_checkpoint(
+    program: Program,
+    checkpoint: Checkpoint,
+    opts: SP1CoreOpts,
+) -> (Vec<ExecutionRecord>, Checkpoint) {
+    let mut runtime = Runtime::recover(program, checkpoint, opts);
+
+    runtime.subproof_verifier = Arc::new(NoOpSubproofVerifier);
+
+    let (events, _) =
+        tracing::debug_span!("runtime.trace").in_scope(|| runtime.execute_record().unwrap());
```

Comment:
I'm unfamiliar with Rust idioms re logging, will this report the file/line it is in? (core/src/utils/prove_distributed/checkpoint.rs:23)

Reply:
No it won't, it is really just a classic `log::debug!(...)` macro but with scoped names.